src/main/java/org/isf/hospital/rest/HospitalController.java

```diff
@@ -0,0 +1,75 @@
+package org.isf.hospital.rest;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.Authorization;
+import org.isf.hospital.dto.HospitalDTO;
+import org.isf.hospital.manager.HospitalBrowsingManager;
+import org.isf.hospital.mapper.HospitalMapper;
+import org.isf.hospital.model.Hospital;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@Api(value = "/hospitals", produces = MediaType.APPLICATION_JSON_VALUE, authorizations = {@Authorization(value = "basicAuth")})
+public class HospitalController {
+
+    private final Logger logger = LoggerFactory.getLogger(HospitalController.class);
+
+    @Autowired
+    protected HospitalBrowsingManager hospitalBrowsingManager;
+    @Autowired
+    private HospitalMapper hospitalMapper;
+
+    public HospitalController(HospitalBrowsingManager hospitalBrowsingManager, HospitalMapper hospitalMapper) {
+        this.hospitalBrowsingManager = hospitalBrowsingManager;
+        this.hospitalMapper = hospitalMapper;
+    }
+
+    @PutMapping(value = "/hospitals/{code:.+}", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
At the moment the application doesn't manage more than one hospital, should we consider only UPDATE api?

Reply:
Do you mean to delete /hospitals (GET) and /hospitals/currencyCode (GET).
How could we retrieve hospitals from api?

Reply:
Sorry, my mistake, I wrongly understood that notation was for an array of `code`